### ISSUE TITLE
Importing PyTorch and ONNX at the same time causes a segmentation fault

### ISSUE THREAD

**Issue by DatCorno** [onnx/onnx#1331]
This issue was first documented [here](https://github.com/apache/incubator-mxnet/issues/12364) by myself. I initially thought the issue lied with mxnet. But I've recreated this error with a simple program using only PyTorch and ONNX.

Most of the information on that page are the same here, but I'll post them again for prosperity.

### Environment
```
----------Python Info----------
Version      : 3.6.5
Compiler     : GCC
Build        : ('default', 'Mar 31 2018 19:45:04')
Arch         : ('64bit', '')
------------Pip Info-----------
Version      : 18.0
Directory    : /home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/pip
----------System Info----------
Platform     : Linux-4.12.14-lp150.12.4-default-x86_64-with-glibc2.3.4
system       : Linux
node         : linux-xdgl
release      : 4.12.14-lp150.12.4-default
version      : #1 SMP Tue May 22 05:17:22 UTC 2018 (66b2eda)
----------Hardware Info----------
machine      : x86_64
processor    : x86_64
Architecture:        x86_64
CPU op-mode(s):      32-bit, 64-bit
Byte Order:          Little Endian
CPU(s):              8
On-line CPU(s) list: 0-7
Thread(s) per core:  2
Core(s) per socket:  4
Socket(s):           1
NUMA node(s):        1
Vendor ID:           GenuineIntel
CPU family:          6
Model:               94
Model name:          Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
Stepping:            3
CPU MHz:             2600.000
CPU max MHz:         3500.0000
CPU min MHz:         800.0000
BogoMIPS:            5184.00
Virtualization:      VT-x
L1d cache:           32K
L1i cache:           32K
L2 cache:            256K
L3 cache:            6144K
NUMA node0 CPU(s):   0-7
Flags:               fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc art arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm 3dnowprefetch cpuid_fault epb invpcid_single pti tpr_shadow vnmi flexpriority ept vpid fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm mpx rdseed adx smap clflushopt intel_pt xsaveopt xsavec xgetbv1 xsaves ibpb ibrs stibp dtherm ida arat pln pts hwp hwp_notify hwp_act_window hwp_epp
----------Network Test----------
Setting timeout: 10
Timing for Gluon Tutorial(en): http://gluon.mxnet.io, DNS: 0.4124 sec, LOAD: 0.6280 sec.
Timing for Gluon Tutorial(cn): https://zh.gluon.ai, DNS: 0.4135 sec, LOAD: 0.5480 sec.
Timing for FashionMNIST: https://apache-mxnet.s3-accelerate.dualstack.amazonaws.com/gluon/dataset/fashion-mnist/train-labels-idx1-ubyte.gz, DNS: 0.7030 sec, LOAD: 0.2876 sec.
Timing for PYPI: https://pypi.python.org/pypi/pip, DNS: 0.0037 sec, LOAD: 0.3840 sec.
Timing for Conda: https://repo.continuum.io/pkgs/free/, DNS: 0.0077 sec, LOAD: 0.0744 sec.
```
### Packages list and version
```
Package           Version       
----------------- --------------
certifi           2018.8.24     
chardet           3.0.4         
graphviz          0.8.4         
idna              2.6           
mxnet             1.3.0b20180824
numpy             1.15.1        
onnx              1.2.2         
Pillow            5.2.0         
pip               18.0          
protobuf          3.6.1         
requests          2.18.4        
setuptools        40.2.0        
six               1.11.0        
torch             0.4.1         
torchvision       0.2.1         
typing            3.6.4         
typing-extensions 3.6.5         
urllib3           1.22          
wheel             0.31.1
```
### Strace near the segmentation fault
```
stat("/home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/onnx", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
stat("/home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/onnx/onnx_cpp2py_export.cpython-36m-x86_64-linux-gnu.so", {st_mode=S_IFREG|0755, st_size=1995800, ...}) = 0
openat(AT_FDCWD, "/home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/onnx/onnx_cpp2py_export.cpython-36m-x86_64-linux-gnu.so", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\320\f\5\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=1995800, ...}) = 0
mmap(NULL, 4101144, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f07afdc6000
mprotect(0x7f07aff52000, 2318336, PROT_NONE) = 0
mmap(0x7f07b0152000, 28672, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x18c000) = 0x7f07b0152000
mmap(0x7f07b0159000, 10104, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f07b0159000
mmap(0x7f07b015c000, 180224, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x194000) = 0x7f07b015c000
mmap(0x7f07b0188000, 163840, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1c0000) = 0x7f07b0188000
close(3)                                = 0
openat(AT_FDCWD, "/home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/onnx/.libs/tls/haswell/x86_64/libprotobuf-487875fe.so.9.0.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/onnx/.libs/tls/haswell/x86_64", 0x7ffd40f83c40) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/onnx/.libs/tls/haswell/libprotobuf-487875fe.so.9.0.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/onnx/.libs/tls/haswell", 0x7ffd40f83c40) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/onnx/.libs/tls/x86_64/libprotobuf-487875fe.so.9.0.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/onnx/.libs/tls/x86_64", 0x7ffd40f83c40) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/onnx/.libs/tls/libprotobuf-487875fe.so.9.0.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/onnx/.libs/tls", 0x7ffd40f83c40) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/onnx/.libs/haswell/x86_64/libprotobuf-487875fe.so.9.0.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/onnx/.libs/haswell/x86_64", 0x7ffd40f83c40) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/onnx/.libs/haswell/libprotobuf-487875fe.so.9.0.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/onnx/.libs/haswell", 0x7ffd40f83c40) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/onnx/.libs/x86_64/libprotobuf-487875fe.so.9.0.1", O_RDONLY|O_CLOEXEC) = -1 ENOENT (No such file or directory)
stat("/home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/onnx/.libs/x86_64", 0x7ffd40f83c40) = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/home/corneau/.virtualenvs/mxnet/lib/python3.6/site-packages/onnx/.libs/libprotobuf-487875fe.so.9.0.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\340\275\22\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=3401800, ...}) = 0
mmap(NULL, 5498952, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f07af887000
mprotect(0x7f07afafe000, 2162688, PROT_NONE) = 0
mmap(0x7f07afcfe000, 61440, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x277000) = 0x7f07afcfe000
mmap(0x7f07afd0d000, 816, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f07afd0d000
mmap(0x7f07afd0e000, 753664, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x287000) = 0x7f07afd0e000
close(3)                                = 0
--- SIGSEGV {si_signo=SIGSEGV, si_code=SEGV_MAPERR, si_addr=0x13b} ---
+++ killed by SIGSEGV (core dumped) +++
[1]    30754 segmentation fault (core dumped)  strace ./example_code.py
```

### Minimum reproducible example
Using the tutorial. Notice the `import onnx` line. The mere presence of it makes the script crash.
```
#!/usr/bin/env python
from torch.autograd import Variable
import torch.onnx
import torchvision
import onnx

dummy_input = Variable(torch.randn(1, 3, 224, 224))

model = torchvision.models.alexnet(pretrained=True)

torch.onnx.export(model, dummy_input, 'alexnet.onnx')
```

**Comment by bddppq** [comment]
We have noticed this issue. A temporary workaround is add "--no-binary onnx" to the pip install command, aka: `pip install --no-binary onnx onnx`.
This will be fixed in the soon coming 1.3 release.

**Comment by DatCorno** [comment]
Alright, thanks a lot!

**Comment by daquexian** [comment]
@bddppq It's not fixed for onnx 1.3.0 on my side.
